platform/darwin/src/MGLFeature.mm

```diff
@@ -28,6 +44,17 @@ - (id)attributeForKey:(NSString *)key {
     return self.attributes[key];
 }
 
+- (NS_DICTIONARY_OF(NSString *, id) *)featureDictionary {
+    return @{@"type": @"Feature",
+             @"properties": (self.attributes) ? self.attributes : @{},
+             @"geometry": @{@"type": @"Point",
```

Comment:
The non-feature direct subclasses of MGLShape should have geoJSONDictionary methods that encapsulate these geometry structures.